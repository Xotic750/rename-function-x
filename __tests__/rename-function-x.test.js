import renameFunction from '../src/rename-function-x';

describe('renameFunction', function() {
  it('is a function', function() {
    expect.assertions(1);
    expect(typeof renameFunction).toBe('function');
  });

  it('should throw when fn not a function', function() {
    expect.assertions(1);
    expect(function() {
      renameFunction(null, 'newname');
    }).toThrowErrorMatchingSnapshot();
  });

  it('should throw if name is invalid', function() {
    expect.assertions(1);
    expect(function() {
      renameFunction(function() {}, 'catch');
    }).toThrowErrorMatchingSnapshot();
  });

  it('rename can be forced', function() {
    expect.assertions(1);
    expect(function() {
      renameFunction(function() {}, 'catch', true);
    }).not.toThrow();
  });

  it('should return a boolean representing success', function() {
    expect.assertions(2);

    const fn = function test1() {};

    const actual = renameFunction(fn, 'newname');

    expect(typeof actual).toBe('boolean');
    expect(fn.name).toBe(actual ? 'newname' : 'test1');
  });
});
