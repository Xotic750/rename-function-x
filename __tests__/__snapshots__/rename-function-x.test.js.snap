// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renameFunction should throw if name is invalid 1`] = `"Not a valid function name \\"new name\\""`;

exports[`renameFunction should throw when fn not a function 1`] = `"null is not a function"`;
